app/assets/javascripts/discourse/app/controllers/composer.js

```diff
@@ -396,6 +396,57 @@ export default Controller.extend({
     return uploadIcon(this.currentUser.staff, this.siteSettings);
   },
 
+  // Use this to open the composer when you are not sure whether it is
+  // already open and whether it already has a draft being worked on. Supports
+  // options to append text once the composer is open if required.
+  //
+  // opts:
+  //
+  // - fallbackToNewTopic: if true, and there is no draft, the composer will
+  // be opened with the create_topic action and a new topic draft key
+  // - insertText: the text to append to the composer once it is opened
+  // - openOpts: this object will be passed to this.open if fallbackToNewTopic is
+  // true
+  @action
+  focusComposer(event, opts = {}) {
```

Comment:
Do we need that event here? 

Maybe in keyboard shortcut, we can stop propagation like
```
focusComposer(event) {
  const composer = this.container.lookup("controller:composer");

  event.preventDefault();
  event.stopPropagation();
  
  composer.focusComposer();
},
```

In that case, we would not need to pass that `null` in tests

Reply:
Yeah that's a good point, I don't need it at all really